File: youxian/src/main/java/apicloud/com/youxian/mvp/presenter/MainPresenter.java
package apicloud.com.youxian.mvp.presenter;


import android.content.Context;

import javax.inject.Inject;

import apicloud.com.youxian.api.AppApi;
import apicloud.com.youxian.common.RxPresenter;
import apicloud.com.youxian.mvp.contract.MainContract;

public class MainPresenter extends RxPresenter<MainContract.View> implements MainContract.Presenter {

    private AppApi appApi;
    private Context context;

    @Inject
    public MainPresenter(AppApi app, Context context) {
        super();
        this.appApi = appApi;
        this.context = context;
    }

}

File: youxian/src/main/java/apicloud/com/youxian/mvp/presenter/MainPresenter.kt
package apicloud.com.youxian.mvp.presenter


import android.content.Context
import apicloud.com.youxian.App
import apicloud.com.youxian.common.RxPresenter
import apicloud.com.youxian.mvp.contract.MainContract
import javax.inject.Inject

class MainPresenter @Inject
constructor(private val app: App, private val context: Context) : RxPresenter<MainContract.View>(), MainContract.Presenter
